code/game/dna/mutations/powers.dm

```diff
@@ -0,0 +1,1200 @@
+	deactivation_messages = list("Everything around you seems to shrink...")
+	instability = GENE_INSTABILITY_MINOR
+	traits_to_add = list(TRAIT_DWARF)
+
+/datum/mutation/midget/New()
+	..()
+	block = GLOB.smallsizeblock
+
+/datum/mutation/midget/activate(mob/M)
+	..()
+	M.pass_flags |= PASSTABLE
+	M.resize = 0.8
+	M.update_transform()
+
+/datum/mutation/midget/deactivate(mob/M)
+	..()
+	M.pass_flags &= ~PASSTABLE
+	M.resize = 1.25
+	M.update_transform()
+
+// OLD HULK BEHAVIOR
+/datum/mutation/hulk
+	name = "Hulk"
+	activation_messages = list("Your muscles hurt.")
+	deactivation_messages = list("Your muscles shrink.")
+	instability = GENE_INSTABILITY_MAJOR
+	traits_to_add = list(TRAIT_HULK, TRAIT_CHUNKYFINGERS)
+	activation_prob = 15
+
+/datum/mutation/hulk/New()
+	..()
+	block = GLOB.hulkblock
+
+/datum/mutation/hulk/activate(mob/M)
+	..()
+	var/status = CANSTUN | CANWEAKEN | CANPARALYSE | CANPUSH
+	M.status_flags &= ~status
+
+/datum/mutation/hulk/deactivate(mob/M)
+	..()
+	M.status_flags |= CANSTUN | CANWEAKEN | CANPARALYSE | CANPUSH
+
+/datum/mutation/hulk/on_draw_underlays(mob/M, g)
+	return "hulk_[g]_s"
+
+/datum/mutation/hulk/on_life(mob/living/carbon/human/M)
+	if(!istype(M))
+		return
+	if(M.health <= 0)
+		M.dna.SetSEState(GLOB.hulkblock, 0)
+		singlemutcheck(M, GLOB.hulkblock, MUTCHK_FORCED)
+		M.update_mutations()		//update our mutation overlays
+		M.update_body()
+		M.status_flags |= CANSTUN | CANWEAKEN | CANPARALYSE | CANPUSH //temporary fix until the problem can be solved.
+		to_chat(M, "<span class='danger'>You suddenly feel very weak.</span>")
+
+/datum/mutation/xray
+	name = "X-Ray Vision"
+	activation_messages = list("The walls suddenly disappear.")
+	deactivation_messages = list("the walls around you re-appear.")
+	instability = GENE_INSTABILITY_MAJOR
+	traits_to_add = list(TRAIT_XRAY_VISION)
+	activation_prob = 15
+
+/datum/mutation/xray/New()
+	..()
+	block = GLOB.xrayblock
+
+/datum/mutation/xray/activate(mob/living/M)
+	..()
+	M.update_sight()
+	M.update_icons() //Apply eyeshine as needed.
+
+/datum/mutation/xray/deactivate(mob/living/M)
+	..()
+	M.update_sight()
+	M.update_icons() //Remove eyeshine as needed.
+
+/datum/mutation/tk
+	name = "Telekenesis"
+	activation_messages = list("You feel smarter.")
+	deactivation_messages = list("You feel dumber.")
+	instability = GENE_INSTABILITY_MAJOR
+	activation_prob = 15
+
+/datum/mutation/tk/New()
+	..()
+	block = GLOB.teleblock
+
+/datum/mutation/tk/on_draw_underlays(mob/M, g)
+	return "telekinesishead_s"
+
+#define EAT_MOB_DELAY 300 // 30s
+
+// WAS: /datum/bioEffect/alcres
+/datum/mutation/sober
+	name = "Sober"
+	activation_messages = list("You feel unusually sober.")
+	deactivation_messages = list("You feel like you could use a stiff drink.")
+
+	traits_to_add = list(TRAIT_ALCOHOL_TOLERANCE)
+
+/datum/mutation/sober/New()
+	..()
+	block = GLOB.soberblock
+
+//WAS: /datum/bioEffect/psychic_resist
+/datum/mutation/psychic_resist
```

Comment:
This one, I mostly kept self contained since psy resist+empath+remote view+mind reading+remote talk all kinda interact with each other and are self contained.